### DISCUSSION TITLE
Regularization parameter for ridge regression penalty in CoxPHSurvivalAnalysis

### DISCUSSION THREAD

**Discussion by chang-hu** [sebp/scikit-survival#236] [Q&A]
Thank you for the awesome package! I ran CoxPHSurvivalAnalysis multiple times with different choices of alpha (0, 0.01, 0.1) and didn't observe any differences in the resulting C-Index, so I checked the source code and found out that the penalty term was divided by n_samples, which doesn't look right to me. Perhaps you meant to divide it by n_features (which is also not common in my experience)? Apologies if I missed something!

**Snippet from scikit-survival/sksurv/linear_model/coxph.py, line 188**

```python
class CoxPHOptimizer:
    def nlog_likelihood(self, w):
        # add regularization term to log-likelihood
        return loss + numpy.sum(self.alpha * numpy.square(w)) / (2. * n_samples)
```

**Versions**

```python
import sklearn; sklearn.show_versions()
import sksurv; print("sksurv:", sksurv.__version__)
# import cvxopt; print("cvxopt:", cvxopt.__version__)
# import cvxpy; print("cvxpy:", cvxpy.__version__)
import numexpr; print("numexpr:", numexpr.__version__)
import osqp; print("osqp:", osqp.OSQP().version())
```

System:
    python: 3.7.6 | packaged by conda-forge | (default, Mar 23 2020, 23:03:20)  [GCC 7.3.0]
executable: /opt/conda/bin/python
   machine: Linux-5.10.47-linuxkit-x86_64-with-debian-buster-sid

Python dependencies:
          pip: 20.1
   setuptools: 46.1.3.post20200325
      sklearn: 0.24.1
        numpy: 1.18.4
        scipy: 1.6.0
       Cython: 0.29.17
       pandas: 1.3.2
   matplotlib: 3.2.1
       joblib: 0.14.1
threadpoolctl: 2.1.0

Built with OpenMP: True
sksurv: 0.15.0.post0
numexpr: 2.7.1
osqp: 0.6.2

**Comment by sebp** [comment]
Whether you divide by n_samples or not really doesn't matter, it only 
changes the scale of the alpha and the scale has no real meaning to 
begin with. In general, which range of alpha one should explore depends 
highly on the dataset and their is not good general rule. It could be 
that the range you explored is unsuitable for your data.

As a heuristic, CoxnetSurvivalAnalysis determines the range for alpha by 
searching for an alpha that will set all coefficients to zero (the upper 
bound), and then selects a portion, e.g. 0.01 of that value as the lower 
bound. I'd suggest you use CoxnetSurvivalAnalysis or do something 
similar for CoxPHSurvivalAnalysis.

**Comment by chang-hu** [comment]
Yeah, I was a bit concerned that it might get a bit confusing when communicating results with others. Anyways, thank you for directing me to CoxnetSurvivalAnalysis!
